### DISCUSSION TITLE
Math needed for UI design

### DISCUSSION THREAD

**Discussion by ghost** [community/community#75620] [Programming Help]
### Body

Hi,

I was going to build a website that allows people to learn HTML, CSS and JavaScript and I was wondering what calculations are needed for UI.

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by santoshvandari** [comment]
When building a website, the UI/UX design calculations depend on the specific features and design elements you want to implement. Here are some considerations and calculations you might need to make:

Responsive Design
Typography
Layout
Flexbox and Grid
Image Dimensions
Interactive Elements
Form Design
Viewport and Screen Sizes

**Comment by elderguardian** [comment]
I personally never had to calculate anything more complicated than 100%/4=25% and every time I did, I ended up finding a simpler solution. As long as you are not planning to do any fancy stuff you should not need any special math knowledge. 

Also, based on the description of your project, I don't think any complicated UI is needed since you should only need text, images, code and maybe videos. You are probably better off running some small CSS and maybe some framework for components. The users of your platform will probably also be thankful for a simple and clean design without distractions.

If you are interested in a small prebuilt CSS, have a look at [this](https://github.com/elderguardian/ton).
